src/main/java/org/broadinstitute/consent/http/db/DataAccessRequestDAO.java

```diff
@@ -35,69 +36,94 @@ public interface DataAccessRequestDAO extends Transactional<DataAccessRequestDAO
-            AND LOWER(v.type) = 'final') final_access_vote ON final_access_vote.reference_id = dar.reference_id
-          WHERE dar.draft = false
-          AND final_access_vote.last_vote = TRUE
-          AND (LOWER(dar.data->>'status') != 'archived' OR dar.data->>'status' IS NULL)
-          AND dar.submission_date > now() - interval '1 year'
-      """)
+  @SqlQuery(
+      """
+              SELECT dar.id, dar.reference_id, dar.collection_id, dar.parent_id,
+                dar.user_id, dar.create_date, dar.sort_date, dar.submission_date, dar.update_date,
+                (regexp_replace(dar.data #>> '{}', '\\\\u0000', '', 'g'))::jsonb AS data,
+                dd.dataset_id
+              FROM data_access_request dar
+              INNER JOIN dar_dataset dd ON dd.reference_id = dar.reference_id AND dd.dataset_id = :datasetId
+              INNER JOIN (
+                SELECT DISTINCT e.reference_id, LAST_VALUE(v.vote)
+                OVER(
+                  PARTITION BY e.reference_id
+                    ORDER BY v.createdate
+                    RANGE BETWEEN
+                      UNBOUNDED PRECEDING AND
+                      UNBOUNDED FOLLOWING
+                ) last_vote
+                FROM election e
+                INNER JOIN vote v ON e.election_id = v.electionid AND v.vote IS NOT NULL
+                WHERE e.dataset_id = :datasetId
+                AND LOWER(e.election_type) = 'dataaccess'
+                AND LOWER(v.type) = 'final') final_access_vote ON final_access_vote.reference_id = dar.reference_id
+              WHERE dar.submission_date > now() - interval '1 year'
+              AND final_access_vote.last_vote = TRUE
+              AND (LOWER(dar.data->>'status') != 'archived' OR dar.data->>'status' IS NULL)
+          """)
   List<DataAccessRequest> findApprovedDARsByDatasetId(@Bind("datasetId") Integer datasetId);
 
+  /**
+   * This query finds submitted DARs based on a date range.  This would be useful if we wanted to
+   * send notifications for "expiring" DARs 30 days before expiration and again at 7 days.
+   *
+   * @param begin Oldest submission date, in seconds since the epoch
+   * @param end   Newest submission date, in seconds since the epoch
+   * @return List of submitted DARs within the date range provided.
+   */
+  @UseRowReducer(DataAccessRequestReducer.class)
+  @SqlQuery(
+      """
+              SELECT dar.id, dar.reference_id, dar.collection_id, dar.parent_id,
+                dar.user_id, dar.create_date, dar.sort_date, dar.submission_date, dar.update_date,
+                (regexp_replace(dar.data #>> '{}', '\\\\u0000', '', 'g'))::jsonb AS data,
+                dd.dataset_id
+              FROM data_access_request dar
+              LEFT JOIN dar_dataset dd ON dd.reference_id = dar.reference_id
+              WHERE dar.submission_date BETWEEN SYMMETRIC :begin AND :end
+          """)
+  List<DataAccessRequest> findSubmittedDarsByTimeRange(@Bind("begin") Timestamp begin,
```

Comment:
Why is this API being added? I don't see it used anywhere in this PR other than tests.

Before when I asked about `SYMMETRIC`, my question was really about how do we know that this needs to support out-of-order queries? Without an example of how it will be used in code, it's hard to tell.

Reply:
The ticket includes an AC of "Mechanism to find expired DARs"
This is an example of a "Mechanism to find expired DARs"

Reply:
If the only need is to find expired, could it be done with a query that has a single bound? Adding the service method that uses this API  (e.g. `DataAccessRequestService.findExpiredDars()`) would help me understand how this API works to meet the AC.

Reply:
Exactly.  As discussed at standup this morning, we have a ticket to notify DAR submitters when their DAR will expire in 30 days (from today).  We'll know the starting time and ending time of the query range.  That's the use case I was specifically thinking about.  
Do others exist?  Sure.  
Are they in other tickets?  Yes.
Do I expect this to change?  Maybe.  It's just an example.